packages/next/src/build/webpack/plugins/middleware-plugin.ts

```diff
@@ -207,9 +235,31 @@ function getCreateAssets(params: {
       Object.keys(middlewareManifest.middleware)
     )
 
-    assets[MIDDLEWARE_MANIFEST] = new sources.RawSource(
-      JSON.stringify(middlewareManifest, null, 2)
-    )
+    if (compilerType === 'server') {
+      nodeServerConfig = middlewareManifest.config
+    }
+
+    if (compilerType === 'edge-server') {
+      edgeServerConfig = middlewareManifest.config
+
+      const outputPath =
+        `${!dev && compilerType !== 'edge-server' ? '../' : ''}` +
```

Comment:
can we switch to consts for maintanance purpose like webpack-config or just use one flag like `isEdgeRuntime` to determine